File: leetcode/python/0191. Number of 1 Bits.py
# Time complexity: O(logN)
# Space complexity: O(1)

class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        numOneBits = 0

        for i in range(32):
            if (n & 1) == 1:
                numOneBits += 1

            n >>=1
        return numOneBits

File: leetcode/java/0191. Number of 1 Bits.java
package leetcode.java;

// Time complexity: O(logN)
// Space complexity: O(1)

class Solution {
    // you need to treat n as an unsigned value
    public int hammingWeight(int n) {
        int numOnes = 0;

        while (n != 0) {
            numOnes += (n & 1);
            n = n >>> 1;
        }
        return numOnes;
    }
}